pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -632,5 +636,128 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
 
         applyMatrix(matrix.data(), wires, inverse);
     }
+
+    /**
+     * @brief Apply a mid-circuit measurement.
+     *
+     * @param wires Wires to sample from.
+     * @param postselect Value to postselect (0 or 1) or empty to ignore
+     * @param reset Flag to reset wire to 0
+     */
+    inline auto
+    applyMidMeasureMP(const std::vector<std::size_t> &wires,
+                      const std::vector<std::size_t> &postselect = {},
+                      bool reset = false) -> int {
+        PL_ABORT_IF_NOT(wires.size() == 1,
+                        "MidMeasureMP should have a single wire.")
+        PL_ABORT_IF(postselect.size() > 1,
+                    "MidMeasureMP accepts at most one postselect value.")
+        const int ps = postselect.empty() ? -1 : postselect[0];
+
+        std::vector<PrecisionT> probs_ = probs(wires[0]);
+        auto sample = randomSample(probs_[0]);
+        if (ps != -1 && ps != sample) {
+            sample = -1;
+        } else {
+            collapse(wires[0], sample);
+            if (reset && sample == 1) {
+                applyOperation("PauliX", wires);
+            }
+        }
+        return sample;
+    }
+
+    /**
+     * @brief Set the seed of the internal random generator
+     *
+     * @param seed Seed
+     */
+    void setSeed(const size_t seed) { rng.seed(seed); }
+
+    /**
+     * @brief Sample 0 or 1 for given probabilities
+     *
+     * @param probs Probabilities of 0.
+     */
+    auto randomSample(const PrecisionT prob_0) {
+        std::discrete_distribution<int> distrib({prob_0, 1. - prob_0});
+        return distrib(rng);
+    }
+
+    /**
+     * @brief Calculate probabilities of measuring 0 or 1 on a specific wire
+     *
+     * @param wire Wire
+     * @return std::vector<PrecisionT> Both probabilities of getting 0 or 1
+     */
+    auto probs(const std::size_t wire) -> std::vector<PrecisionT> {
+        // TODO: use Measurements.probs() instead (resolve circular dependency)
+        auto *arr = this->getData();
+        const size_t stride = pow(2, this->num_qubits_ - (1 + wire));
+        const size_t vec_size = pow(2, this->num_qubits_);
+        const auto section_size = vec_size / stride;
+        const auto half_section_size = section_size / 2;
+
+        PrecisionT prob_0{0.};
+        // zero half the entries
+        // the "half" entries depend on the stride
+        // *_*_*_*_ for stride 1
+        // **__**__ for stride 2
+        // ****____ for stride 4
+        for (size_t idx = 0; idx < half_section_size; idx++) {
+            const size_t offset = stride * 2 * idx;
+            for (size_t ids = 0; ids < stride; ids++) {
+                prob_0 += std::norm(arr[offset + ids]);
+            }
+        }
+
+        std::vector<PrecisionT> probs{prob_0, PrecisionT(1.) - prob_0};
+        return probs;
+    }
```

Comment:
As previously discussed this part of the logic can be moved to the measurements class.

Reply:
Yes, it is in fact already in the measurement class, but we could not use it without tricking CMake to use GCC-specific flags. We'll find a way as per our discussion.

Reply:
As `applyMidMeasure` has been moved to python layer, currently not need for `probs()`